### PR TITLE
feat(react-hooks): extend credential providers with format data

### PR DESCRIPTION
**Pull request by amanji**
Signed-off-by: Akiff Manji <akiff.manji@gmail.com>

### CODE REVIEW

File: packages/react-hooks/src/CredentialProvider.tsx [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-  removeRecord,
-  updateRecord,
-  addRecord,
 } from './recordUtils'
 
-const CredentialContext = createContext<RecordsState<CredentialExchangeRecord> | undefined>(undefined)
+const CredentialContext = createContext<CombinedRecordsState<CredentialExchangeRecord> | undefined>(undefined)
 
-export const useCredentials = () => {
+export const useCombinedCredentials = () => {
```
Comment (TimoGlastra) [inline packages/react-hooks/src/CredentialProvider.tsx:20]: Maybe useCredentialsWithFormatData()? Or do you think that is too long? 

combined doesn't really give any context on what is combined

**Review by TimoGlastra — APPROVED** [review]
Nice!

I've updated the title to be `feat` instead of `refactor` as there is a change in functionality, and otherwise a new release won't get triggered for just a refactor

File: packages/react-hooks/src/recordUtils.ts [hunk 2]
```diff
@@ -14,9 +16,20 @@ import { map, filter, pipe } from 'rxjs'
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type BaseRecordAny = BaseRecord<any, any, any>
 type RecordClass<R extends BaseRecordAny> = Constructor<R> & { type: string }
+
+interface CombinedRecord<R extends BaseRecord> {
+  record: R
+  formatData: GetFormatDataReturn<[IndyCredentialFormat]>
```
Comment (blu3beri) [inline packages/react-hooks/src/recordUtils.ts:22]: If we want to only specifically do `IndyCredentialFormat` I think we should name it accordingly. 

I also implemented this locally for some projects and just made the formatData a separate hook.

`const formattedCredential = useCredentialFormatDataById(credentialId)`

or

`const formattedCredential = useCredentialFormatData()`

Which could then optionally take a generic for the format data, i.e. `IndyCredentialFormat`. I could also create a PR with this code (and for proofs.)

**Comment by amanji** [comment]
I think this PR is probably quite out of sync with what's in AFJ today. 

IMHO if you have working code that expands on this and it's not adding unnecessary work for you I'd say maybe we go with that instead.

Let me know. I'd be happy to review once it's ready.

**Comment by berendsliedrecht** [comment]
Opened #231 that supersedes this.

**Pull request closed without merging**